packages/server/src/db/linkedRecords/index.js

```diff
@@ -88,23 +88,24 @@ exports.attachLinkInfo = async (instanceId, records) => {
     records = [records]
     wasArray = false
   }
+  let modelIds = [...new Set(records.map(el => el.modelId))]
   // start by getting all the link values for performance reasons
-  let responses = await Promise.all(
-    records.map(record =>
-      getLinkDocuments({
-        instanceId,
-        modelId: record.modelId,
-        recordId: record._id,
-        includeDocs: IncludeDocs.EXCLUDE,
-      })
+  let responses = [].concat.apply(
```

Comment:
is the `concat` necessary here? Can we not just pass the array?

Reply:
this is just flattening the array, when the promise response is returned it is like `[[Promise Result], [Promise Result]]` as carrying out view queries now and each query returns a promise.

Thats the way I've flatten arrays in the past, normally I would make a utility function, for it however which I might do (although lodash may have a way to do this already!)